Source/SwiftLintBuiltInRules/Rules/Idiomatic/NoMagicNumbersRule.swift

```diff
@@ -74,7 +74,9 @@ struct NoMagicNumbersRule: OptInRule {
             Example("let foo = 1 >> 2"),
             Example("let foo = 2 >> 2"),
             Example("let foo = 2 << 2"),
-            Example("let a = b / 100.0")
+            Example("let a = b / 100.0"),
+            Example("let (lowerBound, upperBound) = (400, 599)"),
+            Example("let a = (5, 10)")
```

Comment:
Let's also have a test for labeled tuple elements and a tuple used somewhere else that triggers the rule.

After that, I promise to accept the change. 😇

Reply:
So I added those examples, and also extended the allowed tuple elements to include string literals, as cases like `let notFound = (404, "Not Found")` seemed reasonable to skip as well.

Reply:
But hold off on merging for right now @SimplyDanny  - I want to checkout the OSS results, which now seem to have suspiciously lower fixes compared to before

Reply:
ok - that looks better. Should be good to go now once CI passes

Reply:
The strings and booleans make me think again. We now have:

```swift
let a = (3, 4)      // ok
let a = (3, "a")    // ok 
let a = (3, b)      // triggers on 3
let a = (3, (4, 5)) // triggers on all numbers
let a = (true, 4)   // ok
let a = (3, 4 + 5)  // triggers on all numbers
```

Does this make sense? Is it consistent and logical? 🤔 We might need to come up with a better set of rules for tuples I'm afraid.

Reply:
So I totally get your point, but actually, all of the above examples would have been violations before, and there were enough reasonably allowable fixed violations in the OSS tests, that I think the change will add some value for some users.

```
let a = (3, b)      // triggers on 3
let a = (3, (4, 5)) // triggers on all numbers
let a = (3, 4 + 5)  // triggers on all numbers
```

will still be violations, but in the OSS code, there were few or none of these.

Users of the above examples would not be put at any disadvantage by this change - they would just need to disable the rule, like they would today.

I could definitely see a case for allowing `let a = (3, (4, 5))`, but if someone wants that to pass, they can always request that we extend the coverage.

Reply:
Fair enough. Let's have it merged then. Good job!